File: backend/backend/src/main/java/rs/ac/uns/ftn/model/ProductWithImages.java
package rs.ac.uns.ftn.model;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.io.InputStream;
import java.util.List;
import java.util.Set;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ProductWithImages {

    private Long id;

    private String code;

    private String name;

    private String description;

    @JsonManagedReference
    private Set<SizeQuantity> sizeQuantities;

    private double price;

    private ProductCategory category;

    private List<String> imagesName;

}

File: frontend/src/app/models/productWithImages.ts
import { ProductCategory } from "./productCategory";
import { SizeQuantity } from "./sizeQuantity";

export interface ProductWithImages {
    id: number;
    code: string;
    name: string;
    description: string;
    sizeQuantities: Set<SizeQuantity>;
    price: number;
    category: ProductCategory;
    imagesName: string[];
    images: string[];
}